### PR TITLE
Upgrade re2 to 20240201 and add gbenchmark 1.8.3 for test support

### PR DESCRIPTION
**Pull request by jslobodzian**
###### Merge Checklist  

**All** boxes should be checked before merging the PR *(just tick any boxes which don't apply to this PR)*
- [x] The toolchain has been rebuilt successfully (or no changes were made to it)
- [x] The toolchain/worker package manifests are up-to-date
- [x] Any updated packages successfully build (or no packages were changed)
- [x] Packages depending on static components modified in this PR (Golang, `*-static` subpackages, etc.) have had their `Release` tag incremented.
- [x] Package tests (%check section) have been verified with RUN_CHECK=y for existing SPEC files, or added to new SPEC files
- [x] All package sources are available
- [x] cgmanifest files are up-to-date and sorted (`./cgmanifest.json`, `./toolkit/scripts/toolchain/cgmanifest.json`, `.github/workflows/cgmanifest.json`)
- [x] LICENSE-MAP files are up-to-date (`./SPECS/LICENSES-AND-NOTICES/data/licenses.json`, `./SPECS/LICENSES-AND-NOTICES/LICENSES-MAP.md`, `./SPECS/LICENSES-AND-NOTICES/LICENSE-EXCEPTIONS.PHOTON`)
- [x] All source files have up-to-date hashes in the `*.signatures.json` files
- [x] `sudo make go-tidy-all` and `sudo make go-test-coverage` pass
- [x] Documentation has been updated to match any changes to the build system
- [x] Ready to merge

---

###### Summary 
The changes upgrades re2 to 20240201 and adds gbenchmark 1.8.3 for test support.  Both changes are targeted at Azure Linux 3.0

###### Does this affect the toolchain?  
**NO**

###### Test Methodology
Ran pTests against both packages and they passed.

### CODE REVIEW

File: SPECS/LICENSES-AND-NOTICES/LICENSES-MAP.md [hunk 1]
```diff
@@ -5,7 +5,7 @@ The CBL-Mariner SPEC files originated from a variety of sources with varying lic
 | CentOS | [MIT](https://www.centos.org/legal/#licensing-policy) | crash-ptdump-command <br> delve <br> fstrm <br> nodejs-nodemon <br> rhnlib <br> rt-setup <br> rt-tests <br> rtctl <br> tuned |
 | Ceph source | [LGPL2.1](https://github.com/ceph/ceph/blob/master/COPYING-LGPL2.1) | ceph |
 | Debian | [MIT](https://opensource.org/licenses/MIT) | prometheus-process-exporter |
```
Comment (mfrw) [inline SPECS/LICENSES-AND-NOTICES/LICENSES-MAP.md:7]: We might want to rebase because of:
- #7505
- #7529

Comment (jslobodzian) [reply SPECS/LICENSES-AND-NOTICES/LICENSES-MAP.md:7]: merged.

File: cgmanifest.json [hunk 2]
```diff
@@ -25144,8 +25154,8 @@
         "type": "other",
         "other": {
           "name": "re2",
-          "version": "20190801",
-          "downloadUrl": "https://github.com/google/re2/archive/2019-08-01.tar.gz"
+          "version": "20240201",
+          "downloadUrl": "https://github.com/google/re2/archive/2024-02-01.tar.gz"
```
Comment (mfrw) [inline cgmanifest.json:25158]:
```suggestion
          "downloadUrl": "https://github.com/google/re2/archive/refs/tags/2024-02-01.tar.gz"
```

**Review by mfrw — COMMENTED** [review]
I think the failing CGManifest check is due to missing `refs`

**Review by mfrw — APPROVED** [review]
LGTM :)

**Pull request merged**